modules/InfypowerDCSupply/main/power_supply_DCImpl.cpp

```diff
@@ -0,0 +1,176 @@
+        });
+
+    // use a working copy
+    auto caps = this->mod->controller.caps;
+
+    // apply possible overrides to the caps from configuration
+    if (this->mod->config.override_max_current >= 0.0) {
+        if (this->mod->config.override_max_current > caps.max_export_current_A) {
+            EVLOG_warning << "NOT overriding capabilities for maximum export/import current with " << std::fixed
+                          << std::setprecision(1) << caps.max_export_current_A << " A since the value is greater then "
+                          << "the reported value by the power module itself with " << std::fixed << std::setprecision(1)
+                          << caps.max_export_current_A << " A";
+        } else {
+            caps.max_export_current_A = this->mod->config.override_max_current;
+            caps.max_import_current_A = this->mod->config.override_max_current;
+            EVLOG_warning << "Applying override to capabilities for maximum export/import current: " << std::fixed
+                          << std::setprecision(1) << caps.max_export_current_A << " A";
+        }
+    }
+    if (this->mod->config.override_max_power >= 0.0) {
+        if (this->mod->config.override_max_power > caps.max_export_power_W) {
+            EVLOG_warning << "NOT overriding capabilities for maximum export/import power with " << std::fixed
+                          << std::setprecision(1) << caps.max_export_power_W << " W since the value is greater then "
+                          << "the reported value by the power module itself with " << std::fixed << std::setprecision(1)
+                          << caps.max_export_power_W << " W";
+        } else {
+            caps.max_export_power_W = this->mod->config.override_max_power;
+            caps.max_import_power_W = this->mod->config.override_max_power;
+            EVLOG_warning << "Applying override to capabilities for maximum export/import power: " << std::fixed
+                          << std::setprecision(1) << caps.max_export_power_W << " W";
+        }
+    }
+
+    // capabilities must be published at least once
+    this->publish_capabilities(caps);
+
+    // FIXME do we need to publish our initial state?
+    this->publish_mode(types::power_supply_DC::Mode::Off);
+
+    // register (and this enable) publishing of voltage and current values
+    this->mod->controller.on_vc_update.connect([this](float& voltage, float& current) {
+        types::power_supply_DC::VoltageCurrent vc;
+        std::ostringstream dbgmsg;
+
+        vc.voltage_V = voltage;
+        vc.current_A = current;
+
+        this->publish_voltage_current(vc);
+
+        dbgmsg << std::fixed << std::setprecision(1) << "U: " << vc.voltage_V << " V, " << std::fixed
+               << std::setprecision(1) << "I: " << vc.current_A << " A";
+
+        if (this->last_vc_dbgmsg != dbgmsg.str())
+            EVLOG_debug << dbgmsg.str();
+
+        this->last_vc_dbgmsg = dbgmsg.str();
```

Comment:
The assignment can be under the condition.
```suggestion
        if (this->last_vc_dbgmsg != dbgmsg.str()) {
            EVLOG_debug << dbgmsg.str();
            this->last_vc_dbgmsg = dbgmsg.str();
        }
```